### PR TITLE
Issue 258 save connection password

### LINKED ISSUE #258: Support for bulk form exports
This is a new feature proposal originally posted by @yanokwa and discussed in [this post on the forum](https://forum.opendatakit.org/t/adding-bulk-export-of-data-to-briefcase-gui/10868):

> I’d like to update the Export user interface to allow the selection of multiple forms, each one with a separate export directory, export type, private key, etc.
> 
> This is an important feature because some implementers (hi noel) have campaigns that need the flexibility to put different exports in different folders (just like the CLI supports) with different permissions, but they need to do it without requiring the complexities of writing a batch/shell script.
> 
> The changes I’m proposing are:
> 
> - Update Export interface to mirror the Pull/Push interface.
>   - Allow users to select some or all forms and then export the selection with a single button press (similar to Pull/Push)
>   - Interface will have a list of Forms to Export with column headers for Selected, Form Name, and Configure
>   - Interface will have other useful column headers (e.g., Last Exported Date, Details) from community feedback
>   - Configure will allow user to select export directory, private key, and start/end date (similar to the current Export tab) for the specific form
>   - Configure will allow user to enable pull before exporting and tries to use persisted pull credentials
> - Update application so all parameters are persisted between applications launches
>   - Sensitive parameters (e.g., passwords) will have a checkbox to save
> I’ll do all the things to get it to ship in Briefcase, but I’d love feedback from Briefcase users if what I’m proposing sounds reasonable.

I've extracted the following list of new small features (in no particular order) to track work:

- [x] New UI mirroring the Pull/Push interface. (#259)
- [x] New form table column: Configure (to set the export dir, pem file and date ranges for a specific form)
- [x] New form table column: Last export date
- [x] Settings (excluding passwords and any other sensitive parameters) must be persisted between app launches
- [ ] New setting: pull before export
- [x] Setting that the user can check to also save passwords and any other sensitive parameters and acknowledge that this is not safe.

Other related stuff:
- [x] #265 
- [x] #271

### PR DESCRIPTION
**Pull request by ggalmazor**
Closes #258 

This PR adds a new checkbox on the Settings tab to get user's permission to save server connection passwords on preferences. This way, the password fields won't be empty between launcher.

This PR tries to solve this by modifying the least possible amount of code and without changing the design and it prepares our work to support the "pull before export" feature for the Export tab.

Note that the pull and push server connection settings are stored separately and setting them for one tab won't set them for the other.

#### What has been done to verify that this works as intended?
Manual tests on Linux and Win10. Can't get my hands on a mac to test it yet, sorry!
@yanokwa, @kkrawczyk123 could you test this one on a Mac plz?

I didn't work on any automated tests because this part of Briefcase hasn't been refactored yet to be easily testable.

#### Why is this the best possible solution? Were any other approaches considered?
I considered refactoring the SettingsPanel to its own vertical and doing the same work we did for the ExportPanel (own `.form` file for IntelliJ's GUI Designer, separation of vertical/reused artifacts) but the code is really entangled with `BriefcasePreference`'s, `FileSystemUtils`'s and `StorageLocation`'s static API and I don't see an easy way to achieving this without a huge redesign with lots of possible side-effects.

#### Are there any risks to merging this code? If so, what are they?
N/A

### CODE REVIEW

**Comment by kkrawczyk123** [comment]
@ggalmazor as you wish, I checked it on all three systems and it's look good.

**Comment by ggalmazor** [comment]
Thanks!

File: src/org/opendatakit/briefcase/ui/SettingsPanel.java [hunk 1]
```diff
@@ -24,83 +22,97 @@
...
+    chkProxy.setSelected(false);
+    chkProxy.addActionListener(new ProxyToggleListener());
+
+    chkParallel.setSelected(BriefcasePreferences.getBriefcaseParallelPullsProperty());
+    chkParallel.addActionListener(new ParallelPullToggleListener());
+
+    chkTrackingConsent.setSelected(BriefcasePreferences.getBriefcaseTrackingConsentProperty());
+    chkTrackingConsent.addActionListener(new TrackingConsentToggleListener(parentWindow.briefcaseAnalytics));
+
+    chkStorePasswordsConsent.setSelected(BriefcasePreferences.getStorePasswordsConsentProperty());
```
Comment (dcbriccetti) [inline src/org/opendatakit/briefcase/ui/SettingsPanel.java]: Maybe add a tooltip text? 
`chkStorePasswordsConsent.setToolTipText("Passwords are saved in plaintext in the local file system");`
I don’t know if we have a policy on using them.

**Comment by ggalmazor** [comment]
Thanks for the review @dcbriccetti! I've updated the checkbox text to what @yanokwa suggested.

**Pull request merged**